index.d.ts

```diff
@@ -1,426 +1,21 @@
-/// <reference types="node" />
```

Comment:
REVIEW NOTE: This diff is huge because I've moved most of the types to lib/agent.d.ts and this file now just creates the `agent` singleton and re-exports some types from agent.d.ts.  A better representation of the diff is the change from old index.d.ts to new agent.d.ts:  https://gist.github.com/trentm/67b13e5c99abf2ca8001a98e7dd7416e